src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -478,32 +477,32 @@ void Replayer<I>::handle_load_local_group_snapshots(int r) {
   }
 
   auto last_local_snap = get_latest_mirror_group_snapshot(m_local_group_snaps);
-  if (last_local_snap != nullptr) {
-    auto ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(
-        &last_local_snap->snapshot_namespace);
-    if (ns->state != cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY) {
-      if (ns->is_orphan()) {
-        dout(5) << "local group being force promoted" << dendl;
-        handle_replay_complete(&locker, 0, "orphan (force promoting)");
-        return;
-      } else if (m_retry_validate_snap ||
-                 last_local_snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE) {
-        m_retry_validate_snap = false;
-        locker.unlock();
-        schedule_load_group_snapshots();
-        return; // previous snap is syncing, reload local snaps to confirm its completeness.
-      }
-    } else { // primary
-      if (last_local_snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE) {
-        dout(10) << "found incomplete primary group snapshot: "
-                 << last_local_snap->id << dendl;
-        locker.unlock();
-        schedule_load_group_snapshots();
-        return;
-      }
+  if (!last_local_snap) {
+    load_remote_group_snapshots();
+    return;
+  }
+
+  auto* ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(
+      &last_local_snap->snapshot_namespace);
+  ceph_assert(ns != nullptr);
```

Comment:
This can be expressed without an assert (`std::bad_variant_access` exception would be thrown on violation):
```suggestion
  const auto& ns = std::get<cls::rbd::GroupSnapshotNamespaceMirror>(
      last_local_snap->snapshot_namespace);
```

Reply:
Ack